src/middlewared/middlewared/plugins/webdav.py

```diff
@@ -271,5 +275,36 @@ class Config:
         asyncio.ensure_future(middleware.call('service.reload', 'webdav'))
 
 
+class WebDAVFSAttachmentDelegate(FSAttachmentDelegate):
+    name = 'webdav'
+    title = 'WebDAV Share'
+    service = 'webdav'
+
+    async def query(self, path, enabled):
+        results = []
+        for share in await self.middleware.call('sharing.webdav.query', [['enabled', '=', enabled]]):
+            if is_child(share['path'], path):
+                results.append(share)
+
+        return results
+
+    async def get_attachment_name(self, attachment):
+        return attachment['name']
+
+    async def delete(self, attachments):
+        for attachment in attachments:
+            await self.middleware.call('datastore.delete', 'sharing.webdav_share', attachment['id'])
+
+        await self._service_change('webdav', 'reload')
```

Comment:
Wouldnt we have to stop in this case?

Reply:
We don't stop services when exporting pools because they could be serving other pools.

Reply:
Ah, of course, thanks for clarifying